client/my-sites/subscribers/components/no-search-results/no-search-results.tsx

```diff
@@ -0,0 +1,53 @@
+import { Button } from '@automattic/components';
+import { translate } from 'i18n-calypso';
+import { useSelector } from 'calypso/state';
+import { getSiteTitle } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import './style.scss';
+
+type NoSearchResultsProps = {
+	searchTerm: string;
+	setShowAddSubscribersModal: React.Dispatch< React.SetStateAction< boolean > >;
```

Comment:
Personally, I would introduce `AddSubscribersModalContext` for stroing the open state of the modal, and instead of passing down the `setShowAddSubscribersModal` to different components, I would just create a hook for getting the context like so: 
```ts
const { setShow } = useAddSubscribersModal();
```

At the same time, said so, I am not asking you to re-write the implementation. More than that, I just want to share an alternative approach so it is taken to consideration for future work. 🙂

Reply:
Thank you for your review and suggestion, Maciej! I appreciate it!

I was actually considering that, but then when I noticed that in `client/my-sites/subscribers/main.tsx` the `setShowAddSubscribersModal( false );` is called outside of `SubscribersPageProvider` ([over here](https://github.com/Automattic/wp-calypso/pull/78909/files#diff-42a729093f6014ece5f617374dbb7f0d069f745c0ed29596426493141bb0aecfR39)), it made me drop that approach.

But now, looking at this again after reading your comment, I could have just moved `addSubscribersCallback` into the context as well.

I will make a change in another PR so the existing work can go in already.

Reply:
https://github.com/Automattic/wp-calypso/pull/79175 that addresses the refactor is now merged and deployed.